chore: add context to login attempt file and tests

yangkb09 opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
Add context to login attempt file and tests

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Ref #36734

**Special notes for your reviewer**:

yangkb09 committed:
chore: add context to login attempt file and tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=41210) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=41210) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=41210) it.</sub>

Elfo404 committed:
Elasticsearch: Add time zone setting to Date Histogram aggregation (#40882)

* TimeZonePicker: Allow specifying internal timezones to display

* Elasticsearch: Add time zone setting to Date Histogram aggregation

* rename time_zone to timeZone

* Add tests

* fix typo

* Update packages/grafana-data/src/datetime/timezones.ts

Co-authored-by: Chris Cowan <chris@chriscowan.us>

Co-authored-by: Chris Cowan <chris@chriscowan.us>

sarahzinger committed:
Azure Monitor: Add Resource Picker to Template Variable Query Editor (#40841)

Azure Monitor: Add Resource Picker to Template Variable Query Editor
- Should fix bug related broken template variables that relied on a deprecated default workspace.

yuri-tceretian committed:
Alerting: Rule evaluation loop refactoring (#40238)

* move evaluation function out of loop
* extract updateRule function
* isolate alertRule change. update returns new rule and the evaluation accepts the rule as argument
* extract retry loop into a function
* add function wide log context.
* refactor metrics + add tests + replace timeNow with schedule.clock

ashharrison90 committed:
Update Select.mdx (#41205)

* Update Select.mdx

* kick drone

hugohaggmark committed:
Chore: fixes DataSourceRef for adhoc variables (#41232)

* Chore: fixes DataSourceRef for adhoc variables

* Chore: fix path to packages

renovate[bot] committed:
Update dependency fast-json-patch to v3 (#40886)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

ilyastoli committed:
CloudWatch: Add AWS Transfer metrics and dimension (#41168)

What this PR does / why we need it:
Adding AWS Transfer metics according to:
https://docs.aws.amazon.com/transfer/latest/userguide/monitoring.html#metrics

Which issue(s) this PR fixes:
No issue

Special notes for your reviewer:
Based on AWS documentation only, I didn't tried to monitor AWS Transfer myself

renovate[bot] committed:
Update dependency http-server to v14 (#41204)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @types/marked to v3 (#41234)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

oscarkilhed committed:
Barchart/Time series: Allow x axis label (#41142)

idafurjes reviewed (changes requested):
While running grafana on this branch I am getting the message that for the query GetUserLoginAttemptCountQuery Dispatch needs to be replaced with DispatchCtx. As visible in the picture, it would have been nice to have it on the same PR, otherwise great job.

<img width="712" alt="Screenshot 2021-11-03 at 10 09 04" src="https://user-images.githubusercontent.com/36131195/140034012-7072f02e-ff6b-4b2c-9e1f-7ebbaa78cb9d.png">

idafurjes commented:
The same is true for DeleteOldLoginAttemptsCommand
<img width="705" alt="Screenshot 2021-11-03 at 10 17 07" src="https://user-images.githubusercontent.com/36131195/140035612-ad80a6f9-780b-4de0-afa8-6813123d5a14.png">

ilyastoli committed:
CloudWatch: Adde AWS Global Accelerator metrics and dimensions (#41236)

gabor committed:
prometheus: query-field: monaco: implement open-help-by-default (#41112)

* prometheus: query-field: monaco: implement open-help-by-default

* better comment

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* fixed typo in comment

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* better log-message

* fixed comment spelling

Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

* better setting

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>
Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

gabor committed:
prometheus: monaco: autocomplete: offer function-names in functions (#41212)

gabor committed:
prometheus: monaco: allow listing label names + values without a metric (#41233)

* prometheus: monaco: allow listing label names + values without a metric

* removed comment

idafurjes committed:
Chore: Add context to org (#40685)

* Add context to org

* Rebase

* Fix rebase

zoltanbedi committed:
Graph panel: listen to events from timeseries panel (#41033)

* TimeSeries: share cursor with old graph

* Typo fix

* Fix typecheck errors

* Revert "TimeSeries: share cursor with old graph"

This reverts commit 1e7dd4f6270612ceb391f8ae9687ad4b64a5ac7e.

* Graph: subscribe to DataHoverEvent

* Revert "Fix typecheck errors"

This reverts commit 737c8ea8c1f972fd806061842b2985b8dac96214.

joshhunt committed:
Dashboard: Disable image renderer link for unsaved dashboards (#41249)

renovate[bot] committed:
Update dependency inquirer to v8 (#41240)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

hugohaggmark committed:
Chore: Fix for query variables using DataSourceRef (#41255)

AgnesToulet committed:
Navigation: remove redundant config nav link (#41252)

idafurjes committed:
Chore: Add alert ctx (#41161)

* Add context for alert

* Remove context.TODO

* Remove xorm

* Remove context.TODO

* Fix UsageStatsQuerier interface

jsws committed:
NewsPanel: Prevent small titles being split onto multiple lines (#41193)

adeverteuil committed:
Update opt-in.md (#41216)

Need to change that `true` to `false`.

undef1nd committed:
Settings: Add a method to Provider interface to check if a feature is enabled (#41250)

* Settings: Add method to Provider interface to get feature toggles

* Apply review comment

dsotirakis committed:
ReleaseNotes: Update release notes for v8.2.3 (#41269)

* ReleaseNotes: Updated release notes for v8.2.3

* Update release-notes-8-2-3.md

* Update CHANGELOG.md

renovate[bot] committed:
Update dependency jest-junit to v13 (#41259)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

aocenas committed:
Tempo: Add links to nodes in Service Graph pointing to Prometheus metrics (#41135)

sarahzinger committed:
Fix bug that was breaking aws-sdk ConnectionConfig (#41215)

Fix bug that was breaking aws-sdk ConnectionConfig

Commit:
Chore: add context

Commit:
Chore: add context to login and login tests
